packages/headless/src/features/history/history-slice.ts

```diff
@@ -132,4 +88,7 @@ const isPipelineEqual = (current: string, next: string) => current === next;
 
 const isSearchHubEqual = (current: string, next: string) => current === next;
 
+const isFacetOrderEqual = (current: string[] | null, next: string[] | null) =>
+  current === next || (current && next && arrayEquals(current, next));
```

Comment:
In which scenario would `current === next`, as in them being the exact same object reference ?

I guess if both are `null` ? 🤔 

If we make it so the initial state is never `null`, but an empty array, could we skip that and just check the second part of the condition here ?

Reply:
Good point. In the context of `history`, I wouldn't expect a reference to the same array. Along with the unnecessary nullable, I'll remove this condition.